### DISCUSSION TITLE
what is difference between setConsumeTimeout and setTTL

### DISCUSSION THREAD

**Discussion by ljx0517** [weyoss/redis-smq#122] [Q&A] [answered]
I found the method  setConsumeTimeout and setTTL, what is the difference?

**Comment by weyoss** [comment, marked as answer]
@ljx0517 Thank you for opening this issue and for using RedisSMQ.

message.setConsumeTimeout() - This method is useful for setting up a given timeout when a message is being consumed by a consumer. If the timeout exceeds the message processing is cancelled and the message, being processed, is automatically unacknowledged.

message.setTTL() - Allows a message to expire if it has not been delivered to a consumer within a given amount of time.
